Serving tiles, Debian 13

SomeoneElseOSM opened the pull request:
This is raster tiles only (all the serving tiles guides are so far).

It is for the "pgsql" output of osm2pgsql, to match what is on osm.org. A "flex" version (separate, but using much of the same content) would also make sense.

It's a standalone page currently; it'll get added to the menus later. This is so that people can test it and highlight any errors for correction before it becomes part of the main site.

Commit:
Serving tiles, Debian 13
This is raster tiles only (all the serving tiles guides are so far).

It is for the "pgsql" output of osm2pgsql, to match what is on osm.org.
A "flex" version (separate, but using much of the same content) would also make sense.

It's a standalone page currently; it'll get added to the menus later.
This is so that people can test it and highlight any errors for correction before it becomes part of the main site.

pnorman commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+
+## Database functions
+
+In version 5.9.0 of “OSM Carto” (released October 2024), some functions need to be loaded into the database manually. These can be added / re-loaded at any point using:
```
See comment above for versions

SomeoneElseOSM replied:
I've removed "re-loaded"

pnorman reviewed (commented):
Not a full review - I haven't tested anything.

We've historically only used inline code formatting for commands to run and verbatim shell output. This also uses them for other words like usernames, databases, and software names. It's not consistent in this, e.g. mod_tile is sometimes subject to normal grammar rules and sometimes formatted as code.

pnorman commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+git clone https://github.com/gravitystorm/openstreetmap-carto
+cd openstreetmap-carto
+git pull --all
+git switch --detach v5.9.0
```
this can be combined into the git clone command. should we do that?

pnorman commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+
+### Fonts
+
+In version v5.6.0 and above of Carto, fonts need to be installed manually:
```
Also, the version is an OpenStreetMap Carto version, not Carto.

SomeoneElseOSM replied:
I've removed the "version ..." text.

SomeoneElseOSM commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+Next, point a web browser at `http://your.server.ip.address/index.html` (change `your.server.ip.address` to your actual server address). You should see "Apache2 Debian Default Page".
+
+!!! tip
+    If you don't know what IP address it will have been assigned, you can likely use `ifconfig` to find out – if the network configuration is not too complicated it'll probably be the `inet addr` that is not `127.0.0.1`.
```
Changed to `ip address`.  "net-tools" is in debian13-deps.txt and hence installed at this stage, but only available to root.

SomeoneElseOSM commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+!!! tip
+    If you don't know what IP address it will have been assigned, you can likely use `ifconfig` to find out – if the network configuration is not too complicated it'll probably be the `inet addr` that is not `127.0.0.1`.
+
+If you're using a server at a hosting provider then it's likely that your server's internal address will be different to the external address that has been allocated to you, but that external IP address will have already been sent to you, and it'll probably be the one that you're accessing the server on currently.
```
I've changed "likely" to "possible".

Commit:
Changes following review at
https://github.com/switch2osm/switch2osm/pull/299

Commit:
Changed form of words to address comment
https://github.com/switch2osm/switch2osm/pull/299#discussion_r2280526264

SomeoneElseOSM commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+
+It consists of 5 main components: `mod_tile`, `renderd`, `mapnik`, `osm2pgsql` and a `postgresql/postgis` database. Mod_tile is an apache module that serves cached tiles and decides which tiles need re-rendering&nbsp;– either because they are not yet cached or because they are outdated. Renderd provides a priority queueing system for different sorts of requests to manage and smooth out the load f ...
+
+These instructions are have been written and tested against a {{ dist }} server that was essentially newly installed. If you have got other versions of some software already installed (perhaps you upgraded from an earlier version some time ago, or you set up some PPAs to load from) then you may need to make some adjustments.
```
Debian 13 isn't available at the hoster I use yet, so I literally can't test "newly installed" on any sensible hardware.  I've changed the form of words to "on which all this software was newly installed" which still gets the point across but does not claim things that I didn't do.

pnorman commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+
+It consists of 5 main components: `mod_tile`, `renderd`, `mapnik`, `osm2pgsql` and a `postgresql/postgis` database. Mod_tile is an apache module that serves cached tiles and decides which tiles need re-rendering&nbsp;– either because they are not yet cached or because they are outdated. Renderd provides a priority queueing system for different sorts of requests to manage and smooth out the load f ...
+
+These instructions are have been written and tested against a {{ dist }} server on which all this software was newly installed. If you have got other versions of some software already installed (perhaps you upgraded from an earlier version some time ago, or you set up some PPAs to load from) then you may need to make some adjustments.
```
I think this is still to complicated. How about
> These instructions are have been written and tested against a new {{ dist }} server.

pnorman replied:
🤦 I must have been tired when I said "are have been written". That should be "are written"

SomeoneElseOSM replied:
I've gone with "have been written".  Since last week I have now tested most of this on a fresh server.

Commit:
As suggested,
These instructions are have been written and tested against a new {{ dist }} server.

pnorman reviewed (approved):
I haven't tested the changes on Debian 13, but it looks good enough to release

Andygol commented on docs/en/serving-tiles/manually-building-a-tile-server-debian-13.md:
```diff
@@ -0,0 +1,414 @@
+
+The location of the XML file `/home/accountname/src/openstreetmap-carto/mapnik.xml` will need to be changed to the actual location on your system. You can change `[s2o]` and `URI=/hot/` as well if you like. If you want to render more than one set of tiles from one server you can - just add another section like `[s2o]` with a different name referring to a different map style. If you want it to ref ...
+
+The location of the Mapnik plugins directory now varies archtecture.  You can type `` 
```
```suggestion
The location of the Mapnik plugins directory now varies architecture.  You can type
```

SomeoneElseOSM replied:
Thanks - I've fixed that (reworked the sentence to be better English), and also mentioned that you need to run "updatedb" before running locate.

pnorman commented:
> This pull request seems to be missing some of the changes in the next files that provide information about serving tiles on Debian 13 on the site.

I'm confused by this - are you saying those files should be also be changed, or that there's a problem with what the PR is based on?

Commit:
Spelling ("archtecture") and you need to run updatedb first.

Commit:
Grammar: "These instructions have been written"

Commit:
Update inex and mkdocs.yml
Normally I'd make a page "live" before adding to index, but this has had plenty of testing now.
